packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -0,0 +1,215 @@
+      for (const module of Object.entries(modules)) {
+        const moduleFileName = module[0];
+        pluginOutputContext.Modules![moduleFileName] = {
+          Path: generateBicepModuleFilePath(moduleFileName),
+        };
+      }
+    }
+
+    if (outputs) {
+      for (const output of Object.entries(outputs)) {
+        const outputKey = output[0];
+        const outputValue = output[1];
+        pluginOutputContext.Outputs![outputKey] = outputValue;
+      }
+    }
+
+    this.PluginOutput[pluginName] = pluginOutputContext;
+  }
+}
+
+// Stores the bicep orchestration information for all resource plugins
+class BicepOrchestrationTemplate {
+  private ParameterTemplate: string = solutionLevelParameters;
+  private VariableTemplate = "";
+  private ModuleTemplate = "";
+  private OutputTemplate = "";
+  private ParameterJsonTemplate: Record<string, unknown> = {};
+  private RenderContenxt: ArmTemplateRenderContext;
+  private TemplateAdded = false;
+
+  constructor(pluginNames: string[]) {
+    this.RenderContenxt = new ArmTemplateRenderContext(pluginNames);
+  }
+
+  public addTemplate(pluginName: string, scaffoldResult: ScaffoldArmTemplateResult): void {
+    this.ParameterTemplate += this.normalizeTemplateSnippt(
+      scaffoldResult.Orchestration.ParameterTemplate?.Content
+    );
+    this.VariableTemplate += this.normalizeTemplateSnippt(
+      scaffoldResult.Orchestration.VariableTemplate?.Content
+    );
+    this.ModuleTemplate += this.normalizeTemplateSnippt(
+      scaffoldResult.Orchestration.ModuleTemplate?.Content
+    );
+    this.OutputTemplate += this.normalizeTemplateSnippt(
+      scaffoldResult.Orchestration.OutputTemplate?.Content
+    );
+    // update context to render the template
+    this.RenderContenxt.addPluginOutput(pluginName, scaffoldResult);
+    // Update parameters for bicep file
+    Object.assign(
+      this.ParameterJsonTemplate,
+      scaffoldResult.Orchestration.ParameterTemplate?.ParameterJson
+    );
+  }
+
+  public renderOrchestrationFileContent(): string {
+    let orchestrationTemplate = "";
+    orchestrationTemplate += this.normalizeTemplateSnippt(this.ParameterTemplate, false);
+    orchestrationTemplate += this.normalizeTemplateSnippt(this.VariableTemplate, false);
+    orchestrationTemplate += this.normalizeTemplateSnippt(this.ModuleTemplate, false);
+    orchestrationTemplate += this.normalizeTemplateSnippt(this.OutputTemplate, false);
+
+    return compileHandlebarsTemplateString(orchestrationTemplate, this.RenderContenxt);
+  }
+
+  public getParameterFileContent(): string {
+    const parameterObject = {
+      $schema: "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+      contentVersion: "1.0.0.0",
+      parameters: this.ParameterJsonTemplate,
+    };
+    return JSON.stringify(parameterObject, undefined, 2);
+  }
+
+  public needsGenerateTemplate(): boolean {
+    return this.TemplateAdded;
+  }
+
+  private normalizeTemplateSnippt(
```

Comment:
Typo here. Should be `normalizeTemplateSnippet`

Reply:
thanks, fixed